### PR TITLE
Optionally use llapi_rmfids if it's available

### PR DESCRIPTION
**Pull request by utopiabound**
Issue #926

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

### CODE REVIEW

**Comment by utopiabound** [comment]
Manual testing shows this working as expected.
- Calls `llapi_rmfid` if present
- Falls back to `unlink` if it isn't or if llapi_rmfid fails for some other reason (for instance error 524 LUSTRE_ENOTSUPP if rmfid not supported by server)

File: liblustreapi/src/lib.rs [hunk 1]
```diff
@@ -265,29 +313,84 @@ impl Llapi {
...
+
+        let devptr = match CString::new(mntpt) {
+            Ok(c) => c.into_raw(),
+            Err(_) => return Err(flist),
+        };
+
+        let mut fids: Vec<RmfidsArg> = Vec::with_capacity(flist.len() + 1);
+        fids.push(FidArrayHdr::new(0).into());
+        let flist2 = flist.clone();
+        for fidstr in flist {
```
Comment (johnsonw) [inline liblustreapi/src/lib.rs]: One thing to keep in mind is that for loops can also be written as fold's: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.fold

**Comment by johnsonw** [comment]
Just a small comment.

File: liblustreapi/src/lib.rs [hunk 2]
```diff
@@ -70,6 +71,53 @@ impl From<types::lu_fid> for Fid {
...
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub union RmfidsArg {
+    pub hdr: FidArrayHdr,
+    pub fid: Fid,
+}
+
+impl From<FidArrayHdr> for RmfidsArg {
+    fn from(hdr: FidArrayHdr) -> Self {
+        Self { hdr: hdr }
```
Comment (johnsonw) [inline liblustreapi/src/lib.rs]:
```
Self {
   hdr
}
```

File: liblustreapi/src/lib.rs [hunk 3]
```diff
@@ -70,6 +71,53 @@ impl From<types::lu_fid> for Fid {
...
+    }
+}
+
+impl From<Fid> for RmfidsArg {
+    fn from(fid: Fid) -> Self {
+        Self { fid: fid }
```
Comment (johnsonw) [inline liblustreapi/src/lib.rs]:
```
Self {
   fid
}
```

File: liblustreapi/src/lib.rs [hunk 4]
```diff
@@ -265,29 +313,83 @@ impl Llapi {
...
+        if rc == 0 {
+            Ok(())
+        } else {
+            log::info!(
+                "Call to llapi_rmfids({}, [{}, ...]) failed: {}",
+                &mntpt,
+                unsafe { fids[0].hdr.nr },
+                rc
+            );
+            Err(())
```
Comment (johnsonw) [inline liblustreapi/src/lib.rs]: Should we return information in the Err object?

Comment (utopiabound) [reply liblustreapi/src/lib.rs]: No, because we ignore the error and use unlink instead.

**Pull request merged**